### PR TITLE
ci: replace make -C tools with devcluster.

### PR DESCRIPTION
**Pull request by ioga**
## Description

- Replace the last few parts of circleci that use `make -C tools` with `devcluster`.
- Remove `make -C tools run`.

## Test Plan

CI is green.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: tools/Makefile [hunk 1]
```diff
@@ -28,10 +14,6 @@ prep-root:
 	@ln -s $(abspath ../webui/react/build/*) build/webui/react
 	@ln -s $(abspath ../harness/dist/*.whl) build/wheels/
 
-.PHONY: run
```
Comment (rb-determined-ai) [inline tools/Makefile:31]: Why not leave `make -C tools run` intact for a while?  I think rigidly forcing devs to transition from one tooling system to another one is an unnecessary burden.

I think there should be at least a couple months after I land `make devcluster` and before we remove `make -C tools run`

**Comment by hamidzr** [comment]
sorta related: there is one more place we have a copy of the tools dir and that's used for running the tests locally. in `webui/tests/test-cluster`. We can leave that for another PR

**Comment by ioga** [comment]
> sorta related: there is one more place we have a copy of the tools dir and that's used for running the tests locally. in `webui/tests/test-cluster`. We can leave that for another PR

Alright, I want get rid of it in this PR, so we are done with this.

**Comment by hkang1** [comment]
> > sorta related: there is one more place we have a copy of the tools dir and that's used for running the tests locally. in `webui/tests/test-cluster`. We can leave that for another PR
> 
> Alright, I want get rid of it in this PR, so we are done with this.

🙏  ditto on this

**Review by hkang1 — COMMENTED** [review]
very excited for this work to land (love the branch name :))

**Comment by ioga** [comment]
@hamidzr @hkang1 
Okay, I did the `webui/tests/test-cluster`. Can you please check out this PR and try running it the way you use it on your systems? There was some wild path and env magic in `e2e-tests.py` I had to update. I am not super sure about it.

**Review by hamidzr — COMMENTED** [review]
it's great to see that we don't need to depend on package-push job before running e2e tests.  
the main way I run e2e tests locally is through `make test` or `python ./bin/e2e-tests.py e2e-tests` . 
there seems to be some issues with the devcluster config. it's complaining about db.image_name.

also, do I need to set an env variable? DOCKER_LOCALHOST? it'd be nice if the user could run local tests without having to figure out their docker  networking settings.

```
python ./bin/e2e-tests.py e2e-tests
setting up the cluster..
--oneshot is only supported in server mode
Traceback (most recent call last):
  File "./bin/e2e-tests.py", line 93, in det_cluster
    cluster_process = setup_cluster(config, f)
  File "./bin/e2e-tests.py", line 78, in setup_cluster
    raise Exception(f"cluster {config['DET_MASTER']} is unreachable")
Exception: cluster http://localhost:8081 is unreachable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./bin/e2e-tests.py", line 203, in <module>
    main()
  File "./bin/e2e-tests.py", line 199, in main
    fn(config)
  File "./bin/e2e-tests.py", line 130, in e2e_tests
    with det_cluster(config):
  File "/usr/lib/python3.6/contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "./bin/e2e-tests.py", line 96, in det_cluster
    teardown_cluster(config, cluster_process.pid)
UnboundLocalError: local variable 'cluster_process' referenced before assignment
make: *** [Makefile:19: test] Error 1
Time: 0h:01m:17s
```

**Comment by ioga** [comment]
> it's great to see that we don't need to depend on package-push job before running e2e tests.
> the main way I run e2e tests locally is through `make test` or `python ./bin/e2e-tests.py e2e-tests` .
> there seems to be some issues with the devcluster config. it's complaining about db.image_name.

can you please update to a recent devcluster? it had a number of changes recently and I wonder if you're just on an old version.

> also, do I need to set an env variable? DOCKER_LOCALHOST? it'd be nice if the user could run local tests without having to figure out their docker networking settings.

I never heard about `DOCKER_LOCALHOST` so not sure what you're asking about. This thing needs devcluster and docker working, but shouldn't require any other advanced setup.

File: webui/tests/bin/e2e-tests.py [hunk 2]
```diff
@@ -158,12 +153,9 @@ def get_config(args):
     config["CLUSTER_NAME"] = f"det_test_{args.det_port}"
     config["DET_MASTER"] = f"{args.det_host}:{args.det_port}"
 
-    env = {}
```
[the remark this answers is not in the document]
Comment (ioga) [reply webui/tests/bin/e2e-tests.py:161]: devcluster rebuilds the master so it's up-to-date; that needs GOPATH/GOBIN setup properly. This means we'd need to add some new variables to make it work, but everyone's setup is different. 

My idea is that if you can run devcluster, this test can just use the same environment and also work. 

If some env variables are explicitly harmful, maybe we should just filter them out?

**Comment by hamidzr** [comment]
> can you please update to a recent devcluster? it had a number of changes recently and I wonder if you're just on an old version.

let me do that. 

the DOCKER_LOCALHOST popped up somewhere in the logs can't remember where... I asked that because I remember as part of the devcluster setup  I needed to provide it with the interface IP to my docker bridge which is something specific to user setup.

**Comment by ioga** [comment]
ok, let me know.
that `$DOCKER_LOCALHOST` is devcluster's internal thing it does, basically just a configuration magic. it's not a docker thing. you don't need to set it explicitly in your env.

File: webui/tests/test-cluster/devcluster.yaml [hunk 3]
```diff
@@ -21,9 +30,13 @@ stages:
...
       config_file:
         master_host: 127.0.0.1
-        master_port: 8080
-        agent_id: agent1
+        master_port: 8081
         container_master_host: $DOCKER_LOCALHOST
+        agent_id: agent3
+        fluent:
+          port: 24226  # default value is 24224
+          container_name: determined-fluent-3
```
Comment (hamidzr) [inline webui/tests/test-cluster/devcluster.yaml:42]: suggestion: pin the agent slot type to be CPU to that the tests run on the same* type of accelerator between local and ci  `        slot_type: cpu`

**Review by hamidzr — APPROVED** [review]
the WebUI e2e tests section looks good to me. You can't run the tests with node version > 14 but that's the same story on master branch as well

File: webui/tests/bin/e2e-tests.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (hamidzr) [reply webui/tests/bin/e2e-tests.py:161]: IMO being explicit about them should minimize potential surprises that we get but either way works

**Comment by hamidzr** [comment]
> can you please update to a recent devcluster? it had a number of changes recently and I wonder if you're just on an old version.

> I asked that because I remember as part of the devcluster setup I needed to provide it with the interface IP to my docker bridge 


ftr updating helped. I'm guessing it's because of the newly added feature to guess the host `--no-guess-host`

**Review by hkang1 — APPROVED** [review]
This is SOOOOOO much better. Especially the removal of `make -C tools` from `webui/tests`! `devcluster` and `make -C webui/tests test` worked great for me. Thanks @ioga!

**Pull request merged**